react/react1/week3/reactweek3/src/TodoList.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from "react";
+import TodoItem from "./TodoItem";
+// import { DataFetching } from "./DataFetching";
+
+export function TodoList() {
+  const [useTodo, setUseTodo] = useState([]);
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const Api =
+    "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+  useEffect(() => {
+    if (setUseTodo) {
+      fetch(Api)
+        .then((res) => res.json())
+        .then((data) => {
+          const fetchMap = data.map((data) => ({
+            id: data.id,
+            description: data.description,
+            deadline: data.deadline,
+          }));
+
+          setUseTodo((prev) => [...prev, ...fetchMap]);
+        });
+    }
+  }, [setUseTodo]);
+
+  // for add State
+  function addTodo() {
+    let addID;
+    if (useTodo.length === 0) {
+      addID = 1;
+    } else {
+      addID = useTodo[useTodo.length - 1].id + 1;
+    }
+    const randomTodo = {
+      id: addID,
+      description: description,
+      deadline: deadline,
+    };
+    console.log(randomTodo);
+    setUseTodo((prev) => [...prev, randomTodo]);
+  }
+
+  const todoValues = useTodo.map((item) => {
+    return (
+      <TodoItem
+        description={item.description}
+        deadline={item.deadline}
+        deleteTodo={deleteTodo}
+        editTodo={editTodo}
+        key={item.id}
+        id={item.id}
+      />
+    );
+  });
+  // for Delete State
+
+  function deleteTodo(id) {
+    return setUseTodo((prevTodos) => {
+      let dataItems = prevTodos.filter((item) => item.id != id);
+      return dataItems;
+    });
```

Comment:
There's no need to return `setUseTodo` here.

Reply:
done